[flyingsalsa] ip

flyingsalsa opened the pull request:
# notGPT 🚀

> "Organize your life, one task at a time." - me

notGPT is your go-to solution for efficient task management. It's:

* User-friendly
* Customizable
* ~~Slow~~ *Lightning* fast

Getting started is easy:

1. Download notGPT [here](https://github.com/flyingsalsa/ip/releases/download/v0.1.1/notGPT.jar)
2. Make sure you have [Java 17 and above](https://www.oracle.com/java/technologies/downloads/#java22) installed
3. Use  ```java -jar notGPT.jar``` in your cmd line to run! (make sure you're in the correct directory)
_*doesnt work if you just click on the program coz it has no gui rn lol_
4. win

## Features

- [x] Create and delete tasks
- [x] Assign dates to tasks
- [ ] Sync across devices (not coming soon)
- [ ] Email reminders (no)

Here's a sneak peek of our main function:

```javascript
public static void main(String[] args) {
            Scanner scanner = new Scanner(System.in);
...
```
 ~~WOW i love scanners~~

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

flyingsalsa committed:
Duke.java: modify for Level-0
Add Parser.java

flyingsalsa committed:
Update Parser.java

flyingsalsa committed:
Update Parser.java to take user input for Level-1

flyingsalsa committed:
Add Storage.java

flyingsalsa committed:
Modify Store.java to use Task Array

flyingsalsa committed:
Modify Parse.java to save Text to Storage

flyingsalsa committed:
Remove Echo from Parser

flyingsalsa committed:
Modify Duke to work with Parser for Level-2

flyingsalsa committed:
Add mark and unmark to Parser for Level-3

flyingsalsa committed:
Merge pull request #1 from flyingsalsa/develop

Merging develop into master

flyingsalsa committed:
Add Todo, Event and Dealine.java

flyingsalsa committed:
Update todo.java to be functional

flyingsalsa committed:
Modify Parser to incorporate Deadline and Event

flyingsalsa committed:
Update logic for Parser for Level-4

flyingsalsa committed:
Add input.txt and expected.txt

flyingsalsa committed:
Add deletion to Parser

flyingsalsa committed:
Change error messages for niche cases

flyingsalsa committed:
Update README.md

flyingsalsa committed:
Update README.md

flyingsalsa committed:
Update Code to conform to Standard

flyingsalsa committed:
Change name of Duke class to Notgpt

flyingsalsa committed:
Reorder Text output from Notgpt startup

flyingsalsa committed:
Modify Storage to update data.txt

flyingsalsa committed:
Add ability to save and load data for Level-7

flyingsalsa committed:
Update .gitignore to ignore data

flyingsalsa committed:
Refactor package organization

flyingsalsa committed:
Modify Tasks to parse for LocalDates

flyingsalsa committed:
Refactor packages

flyingsalsa committed:
Merge branch 'add-gradle-support'

flyingsalsa committed:
Update build.gradle

flyingsalsa committed:
Add test class to test directory

flyingsalsa committed:
Modify Parser for .jar

flyingsalsa committed:
Add find command to Parser

zackjh commented on src/main/java/duke/ui/Parser.java:
```diff
@@ -0,0 +1,129 @@
+                text = parts[1];
+            }
+            switch (command) {
+            case "list":
```
I like how you didn't indent the `case` statements, as per the coding standards. Good job!

zackjh commented on src/main/java/duke/Tasks/Deadline.java:
```diff
@@ -0,0 +1,25 @@
+package duke.Tasks;
+import duke.additionalParsers.DateParser;
+
+
```
Should this have a JavaDoc header comment? I noticed that not all classes have header comments - not sure if it's a requirement, but you might want to take a look into this.

zackjh commented on src/main/java/duke/Tasks/Task.java:
```diff
@@ -0,0 +1,31 @@
+package duke.Tasks;
+public class Task {
+    public String task;
+    private boolean done;
```
You could consider renaming `done` to `isDone` (or something similar) since it's a `boolean`, as per the [coding standard](https://se-education.org/guides/conventions/java/intermediate.html#naming).

zackjh commented on src/main/java/duke/ui/Parser.java:
```diff
@@ -0,0 +1,129 @@
+                    System.out.println("sorry bud that ain't a number\ni don't know which task u're referring to...");
+                    break;
+                }
+                if(0 < i && i <= storage.size()) {
+                    storage.mark(i);
+                    System.out.println(String.format("marked %s as completed\nuse \"list\" to see changes",i));
+                }else {
+                    System.out.println("that number isn't a valid task dude...");
+                    System.out.println("it has to be from 1 to " + storage.size());
+                }
```
Should there be a space after `if` and `else`?

zackjh reviewed (commented):
There are some minor errors with regard to the coding standard but other than that, looks good!

flyingsalsa committed:
Update .gitignore

flyingsalsa committed:
Update .gitignore

flyingsalsa committed:
Merge branch 'branch-Level-8'

flyingsalsa committed:
Delete Data folder from commits

flyingsalsa committed:
Update coding standards

flyingsalsa committed:
Merge branch 'branch-A-CodingStandard'

flyingsalsa committed:
Remove error.log

flyingsalsa committed:
Merge branch 'branch-A-JavaDoc'

flyingsalsa committed:
Modify parser for Level-9

flyingsalsa committed:
Merge branch 'branch-Level-9'

flyingsalsa commented:
thanks for the comments! I will take note for future commits

flyingsalsa committed:
Rename done is Task to isDone

flyingsalsa committed:
Update README.md

flyingsalsa committed:
Update README.md

flyingsalsa committed:
Add whitespaces to conform to coding standard

flyingsalsa committed:
Merge branch 'master' of https://github.com/flyingsalsa/ip

flyingsalsa committed:
Add Checkstyle files

flyingsalsa committed:
Modify build.gradle to run Checkstyle

flyingsalsa committed:
Add javadoc comments

flyingsalsa committed:
Update Storage.java

flyingsalsa committed:
Update Task.java

flyingsalsa committed:
Add more Javadocs

flyingsalsa committed:
Correct coding standards based on checkstyle

flyingsalsa committed:
Update build.gradle

flyingsalsa committed:
Create TextBotApp class to handle JavaFX UI

flyingsalsa committed:
Update Notgpt.java

flyingsalsa committed:
Merge branch 'branch-Level-10'

flyingsalsa committed:
Update TextBotApp to run properly

flyingsalsa committed:
Integrate Parser with TextbotApp

flyingsalsa committed:
Update Class names

MiloTruck commented on src/main/java/duke/additionalparsers/DateParser.java:
```diff
@@ -0,0 +1,39 @@
+ * them to {@link LocalDate} and then reformatting them back to a new {@link String} with corrected formatting.
+ */
+public class DateParser {
+    private List<DateTimeFormatter> formatters = Arrays.asList(
+            DateTimeFormatter.ofPattern("yyyy-MM-dd"),
+            DateTimeFormatter.ofPattern("yyyy.MM.dd"),
+            DateTimeFormatter.ofPattern("dd-MM-yyyy"),
+            DateTimeFormatter.ofPattern("dd.MM.yyyy")
+    );
```
Extra tab spacing here:

```java
    private List<DateTimeFormatter> formatters = Arrays.asList(
        DateTimeFormatter.ofPattern("yyyy-MM-dd"),
        DateTimeFormatter.ofPattern("yyyy.MM.dd"),
        DateTimeFormatter.ofPattern("dd-MM-yyyy"),
        DateTimeFormatter.ofPattern("dd.MM.yyyy")
    );
```

MiloTruck replied:
Extra tab spacing here:

```java
    private List<DateTimeFormatter> formatters = Arrays.asList(
        DateTimeFormatter.ofPattern("yyyy-MM-dd"),
        DateTimeFormatter.ofPattern("yyyy.MM.dd"),
        DateTimeFormatter.ofPattern("dd-MM-yyyy"),
        DateTimeFormatter.ofPattern("dd.MM.yyyy")
    );
```

MiloTruck commented on src/main/java/duke/ui/Parser.java:
```diff
@@ -0,0 +1,138 @@
+     *
+     * @param input system input from textbox
+     */
+    public String parse(String input) {
```
Maybe this should be renamed to `parseInput()`, might be clearer...?

MiloTruck commented on src/main/java/duke/tasks/Storage.java:
```diff
@@ -0,0 +1,192 @@
+ * </p>
+ */
+public class Storage {
+    private ArrayList<Task> store = new ArrayList<>();
```
A better name would be something like `taskList`, so it becomes obvious what the `ArrayList` stores.

MiloTruck commented on src/main/java/duke/ui/NotgptJavaFX.java:
```diff
@@ -0,0 +1,132 @@
+        inputBox.getChildren().addAll(inputField, sendButton);
+
+        // Layout
+        VBox root = new VBox();
```
Maybe renaming this to `vBoxRoot` might make it clearer what `root` is.

MiloTruck commented on src/main/java/duke/ui/NotgptJavaFX.java:
```diff
@@ -0,0 +1,132 @@
+
+public class NotgptJavaFX extends Application {
+
+    private Parser parser = new Parser();
+    private VBox chatBox;
+    private TextField inputField;
+    private Button sendButton;
+    private Image userAvatar;
+    private Image botAvatar;
```
Naming here is pretty good. It is quite obvious what each variable represents without additional comments.

MiloTruck reviewed (commented):
Some naming suggestions.

flyingsalsa committed:
Fix shadowjar

flyingsalsa committed:
Rename variables

flyingsalsa committed:
Update Notgpt

flyingsalsa committed:
Update README.md

flyingsalsa committed:
Add Assertions to Parser

The Parser class lacks internal checks for its assumptions and invariants. This makes the code vulnerable to unexpected behavior and difficult to debug.
Let's add strategic assertions throughout the Parser class. These assertions will validate input, ensure proper initialization of objects, verify state consistency, and catch unhandled cases.
By using assertions, we can catch potential issues early in the development and testing phases. This approach follows best practices for defensive programming and helps document our code's assumptions.
The assertions are placed to cover:

Storage object state
Task index bounds
Post-operation state consistency
Exhaustive switch statement handling

Remember to run with the -ea flag to enable assertions during development and testing.

flyingsalsa committed:
Merge branch 'branch-A-Assertions'

flyingsalsa committed:
Bye command: make bye actually terminate the programe

Bye command is unable to shut down the JavaFx program.

This is a core functionality of the program, to be able to exit on its own without forcing the user to resort to OS methods shut as task manager or click on the "x" to close the window

Let's change the behaviour of the Parser to display the goodbye message and close the program afterwards but also add a delay in order to allow users to read the goodbye message

flyingsalsa committed:
Merge Commit from master

flyingsalsa committed:
Update code to conform to coding standards

flyingsalsa committed:
Merge branch 'branch-A-CodeQuality'

flyingsalsa committed:
Move gradle.yml into appropriate folder

flyingsalsa committed:
Test CI

flyingsalsa committed:
Update gradle.yml

flyingsalsa committed:
Update gradle.yml

flyingsalsa committed:
Update NotgptJavaFX.java

flyingsalsa committed:
Update NotgptJavaFX.java

flyingsalsa committed:
Create DayToDate parser

flyingsalsa committed:
Update DayToDateParser

flyingsalsa committed:
Add JavaDoc for DayToDateParser

flyingsalsa committed:
Merge pull request #4 from flyingsalsa/branch-A-Streams

Update NotgptJavaFX.java

flyingsalsa committed:
Merge branch 'master' into branch-A-CodeQuality

flyingsalsa committed:
Merge pull request #3 from flyingsalsa/branch-A-CodeQuality

Update NotgptJavaFX.java

flyingsalsa committed:
Update NotgptJavaFX.java to conform to checkstyle

flyingsalsa committed:
Update codingStandard

flyingsalsa committed:
Refactor code for DayToDateParser

flyingsalsa committed:
Add Command Classes to comply with OOP standards

flyingsalsa committed:
Reformat Packages

flyingsalsa committed:
Update code to conform with checkstyle

flyingsalsa committed:
Update gradle.yml

flyingsalsa committed:
Update Notgpt.java

flyingsalsa committed:
Update README.md

flyingsalsa committed:
Rename DataParser

flyingsalsa committed:
Update GUI

flyingsalsa committed:
Update GUI

flyingsalsa committed:
Merge remote-tracking branch 'origin/master'

flyingsalsa committed:
Add Ui.png

flyingsalsa committed:
Fix Coding Style

flyingsalsa committed:
Update Gui

flyingsalsa committed:
Fix textbox truncation bug

flyingsalsa committed:
update Ui.png

flyingsalsa committed:
Update DateParser.java

flyingsalsa committed:
Fix code checkstyle requirements

flyingsalsa committed:
Update README.md

flyingsalsa committed:
Create _config.yml

flyingsalsa committed:
Update _config.yml

flyingsalsa committed:
Update README.md

flyingsalsa committed:
Update README.md

flyingsalsa committed:
Update _config.yml

flyingsalsa committed:
Update README.md

flyingsalsa committed:
Update _config.yml

flyingsalsa committed:
Update README.md

flyingsalsa committed:
Update README.md

flyingsalsa committed:
Update _config.yml

flyingsalsa committed:
Update README.md

flyingsalsa committed:
Update README.md

flyingsalsa committed:
Update README.md

flyingsalsa committed:
Update README.md

flyingsalsa committed:
Update README.md

flyingsalsa committed:
Update README.md

flyingsalsa committed:
Update README.md

flyingsalsa committed:
Update README.md

flyingsalsa committed:
Adjust Chatbot Output

flyingsalsa committed:
Merge branch 'master' of https://github.com/flyingsalsa/ip

flyingsalsa committed:
Update README.md

flyingsalsa committed:
Update README.md

flyingsalsa committed:
Update README.md

flyingsalsa committed:
Update NumberCommandTest.java

flyingsalsa committed:
Merge branch 'master' of https://github.com/flyingsalsa/ip

flyingsalsa committed:
Update Ui.png

flyingsalsa committed:
Update README.md

flyingsalsa committed:
Update README.md

flyingsalsa committed:
Update README.md

flyingsalsa committed:
Update README.md

flyingsalsa committed:
Update README.md

flyingsalsa committed:
Update README.md
